components/automate-ui/src/app/components/resource-dropdown/resource-dropdown.component.ts

```diff
@@ -129,13 +158,31 @@ export class ResourceDropdownComponent implements OnInit, OnChanges {
   }
 
   private updateLabel(): void {
-    const checkedResources = this.resources.filter(p => p.checked);
+    const checkedResources = this.allCheckedResourceNames;
     this.label = checkedResources.length === 0 ? this.noneSelectedLabel
-      : checkedResources.length === 1 ? checkedResources[0].name
+      : checkedResources.length === 1 ? checkedResources[0]
         : `${checkedResources.length} ${this.objectNounPlural}`;
   }
 
-  get disabled(): boolean {
-    return this.resources.length === 0;
+  // Context is a *closed* dropdown, so use the raw resources, not the last snapshot
+  private isDisabled(): boolean {
+    return this.resources.length === 0 || this.allResourcesCount === 0;
   }
+
+  private calculateAllFilteredResourcesCount(): number {
+    return this.filteredResources.reduce(
+      (sum, group) => sum + group.itemList.length, 0);
+  }
+
+  private get allResourcesCount(): number {
+    return this.resources.reduce(
+      (sum, group) => sum + group.itemList.length, 0);
+  }
```

Comment:
Hmm would be be possible to define a `length()` or `count()` method on `ResourceCheckedSection[]`?

Reply:
I do not think you can add methods to an interface or type; would have to make it a full-fledged class, so not sure that is worth it. However, I did remove the duplicated code.